### ISSUE TITLE
module_invoke() doesn't seem to load Views blocks

### ISSUE THREAD

**Issue by bobchristenson** [backdrop/backdrop-issues#4119]
I was looking to embed a views block into a template using my standard method that looks something like this:
```
$block = module_invoke('views', 'block_view', 'views_block_delta');    
dpm($block);    
print $block['content'];    
```

Other types of blocks load fine with this method.  But if I try to load a Views block, then the $block var is always empty. 

_PS: I'm not sure whether you use render() to print the block content in backdrop or not, so ignore this line for now, including it only for completeness, I've tried both with and without render()._

**Comment by klonos** [comment]
From Gitter...

@hosef:

> The views module in Backdrop does not implement hook_block_view, so there is nothing to return from module_invoke(). Backdrop has a fancy new Object Oriented block layer basesd on the panels pane API that modules can use instead, which is what views is using.

@laryn:
> @bobchristenson Maybe https://api.backdropcms.org/api/backdrop/core%21modules%21views%21views.module/function/views_embed_view/1 is a viable alternative?

**Comment by klonos** [comment]
...I think that perhaps throwing a watchdog message each time `module_invoke('views', ... );` is called is in order, pointing people to documentation on what has changed, and what the alternative way to do things is.

**Comment by bobchristenson** [comment]
So, I'm using views_embed_view successfully in other situations, so I know that works as a replacement in most cases.  My example probably should have been for the one place you can't use this instead of a views block:  an exposed filter block.

Is there a way other than module_invoke to embed a views exposed filter block?

**Comment by laryn** [comment]
I wasn't aware that you couldn't use `views_embed_view` to embed an exposed filter block -- what happens?

**Comment by bobchristenson** [comment]
I guess I dont know how you'd use views_embed_view to display something that is only provided as a block...To be clear, I'm choosing to provide the exposed filters as a block within the views UI.  I dont think you can use views_embed_view to display this block...or can you (somehow)?

**Comment by laryn** [comment]
Try something like this:

`views_embed_view('machine_name_of_view', 'machine_name_of_block_display');`

You can also pass additional parameters as arguments if needed.

**Comment by bobchristenson** [comment]
Yeah, I think you're misunderstanding.  It looks like this code is still attempting to display the CONTENT of the view.  That's not what I'm trying to do.  Views provides a block that contains the exposed FILTERS for the view (as enabled by what you see in the screenshot attached)
![Screen Shot 2019-10-08 at 2 21 52 PM](https://user-images.githubusercontent.com/282873/66422062-fe494a80-e9d6-11e9-9a4c-e06e61826f6e.png)

**Comment by klonos** [comment]
Related: #1322

**Comment by BWPanda** [comment]
@bobchristenson I was just testing this on my site to see what I could come up with for you, and while I don't have a full solution, maybe this will point you in the right direction...

I added an exposed form to a view, then enabled the exposed form as a block (as per your above screenshot). I then, using Devel module's Execute PHP Code page (/admin/devel/php), got the info for all Views' blocks:

```php
dpm(layout_get_block_info('views'));
```
(see [layout_get_block_info](https://api.backdropcms.org/api/backdrop/core%21modules%21layout%21layout.module/function/layout_get_block_info/1))

In my case, there was only one Views' block - the exposed one (you might have more). The output shows the names and descriptions of the blocks, but also the class: `ViewsSpecialBlock`. When I searched for this on the API site, I found this: [ViewsSpecialBlock::getContent](https://api.backdropcms.org/api/backdrop/core%21modules%21views%21includes%21views.special_block.inc/function/ViewsSpecialBlock%3A%3AgetContent/1)

Now unfortunately my OOP skills aren't that good, but it sounds like this may be what you're looking for... I encourage you to look into using this class/method to find the content of the exposed form block so you can do what you want with it. Hope that helps somewhat!
